Assets/Scripts/Tools/GraphArrangerTool.cs

```diff
@@ -26,15 +29,24 @@ public void OnLeftClick(Transform cameraTransform, bool isRayCastHit, RaycastHit
       _toolPanelController.SetBackgroundColor(GetPanelColor());
     }
 
-    public void OnRightClick(Transform cameraTransform, bool isRayCastHit, RaycastHit raycastHit) =>
+    public void OnRightClick(Transform cameraTransform, bool isRayCastHit, RaycastHit raycastHit)
+    {
       _graphArranger.ToggleMode();
+      RefreshToolgunModeText();
+    }
 
     public void OnLeftMouseButtonHeld(Transform cameraTransform) { }
 
     public void OnLeftMouseButtonReleased() { }
 
-    public void OnSelect() => _toolPanelController.SetBackgroundColor(GetPanelColor());
+    public void OnSelect()
+    {
+      _toolPanelController.SetBackgroundColor(GetPanelColor());
+      RefreshToolgunModeText();
+    }
 
     private Color GetPanelColor() => _graphArranger.ArrangeEnabled ? Color.green : Color.red;
+
+    private string GetArrangerMode() => EnumUtils<GraphArrangerMode>.GetName(_graphArranger.ArrangeMode).Substring(1);
```

Comment:
Why do we cut the first letter?

Reply:
Ah, the _.
I hate this approach as adding more modes will probably break this. But no better idea at this moment.

Reply:
because the first letter in this enum is '_' (_2D, _3D)

Reply:
I would love it if we used    [Display(Name = "3D")] for the enums and use that name somehow. I think it is the best approach.

Reply:
fixed with [Description("")]